codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -218,8 +224,8 @@ public void goAwayReceived(final int lastKnownStream, long errorCode, ByteBuf de
         for (int i = 0; i < listeners.size(); ++i) {
             try {
                 listeners.get(i).onGoAwayReceived(lastKnownStream, errorCode, debugData);
-            } catch (RuntimeException e) {
-                logger.error("Caught RuntimeException from listener onGoAwayReceived.", e);
+            } catch (Throwable cause) {
```

Comment:
@nmittler - For now yes. If we let exceptions propagate from user code our internal state may become corrupted if we have not completely updated it. I submitted a PR a while ago to try and allow exceptions to propagate but it needed more work and I have not got around to updating it (I will likely wait until childchannels comes to be).